src/praisonai/praisonai/gateway/server.py

```
@@ -1241,18 +1241,29 @@ async def approval_resolve(request):
-                return JSONResponse({
-                    "allow_list": _approval_mgr.allowlist.list(),
-                })
+                resp = {"allow_list": _approval_mgr.allowlist.list()}
+                list_scoped = getattr(
+                    _approval_mgr.allowlist, "list_scoped", None
+                )
+                if callable(list_scoped):
+                    try:
+                        resp["grants"] = list_scoped()
+                    except Exception:
+                        pass
```

Ruff (0.15.20)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)